pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -340,7 +340,33 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 	return nil
 }
 
-func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, resourceLister listers.PipelineResourceLister, kubeclient kubernetes.Interface, logger *zap.SugaredLogger) {
+// This function constructTaskStepsSorter constructs a map matching the names of
+// the steps to their indices for a task.
+func constructTaskStepsSorter(taskSpecSteps []corev1.Container) map[string]int {
+	sorter := make(map[string]int)
+	for index, step := range taskSpecSteps {
+		sorter[step.Name] = index
+	}
+	return sorter
+}
+
+// This function sortTaskRunSteps sorts the steps of the task run, based on the
+// order of the steps in the task.
+func sortTaskRunSteps(taskRunSteps []v1alpha1.StepState, taskSpecSteps []corev1.Container) {
+	// Generate the map matching the step name and its index
+	sorter := constructTaskStepsSorter(taskSpecSteps)
+	for index, step := range taskRunSteps {
```

Comment:
We may be able to use the `sort` package here :angel:

Reply:
I tried, but the the sort package is easy to sort the array with a filed or multiple fields in itself. Have not figured out to sort one array by aligning with another array.